File: app/assets/javascripts/ng-app/services/position.js
angular.module('Reserva')
  .factory('Position', 
      [         'ActiveResource',
      function ( ActiveResource){
          
        function Position(){
          this.number('id');
          this.string('name');
          this.number('establishment_id');

          this.belongsTo('establishment');
        };

        Position.inherits(ActiveResource.Base);
        Position.api.set('http://reserva.dev/api/v1');

        return Position;

  }]);

File: app/models/position.rb
# == Schema Information
#
# Table name: positions
#
#  id               :integer          not null, primary key
#  name             :string(255)
#  establishment_id :integer
#  created_at       :datetime
#  updated_at       :datetime
#  deleted_at       :datetime
#
# Indexes
#
#  index_positions_on_deleted_at  (deleted_at)
#

class Position < ActiveRecord::Base
  acts_as_paranoid #implements soft deletion

  validates :name, presence: true

  belongs_to :establishment
  has_many :appointments
end
